cli/build.rs

```diff
@@ -41,21 +44,51 @@ fn main() -> std::io::Result<()> {
     }
     println!("cargo:rerun-if-env-changed=NIX_JJ_GIT_HASH");
 
-    if let Some(git_hash) = get_git_hash() {
-        println!("cargo:rustc-env=JJ_VERSION={}-{}", version, git_hash);
+    if let Some((mut git_hash, maybe_date)) = get_git_timestamp_and_hash() {
+        git_hash.truncate(16);
+        println!(
+            "cargo:rustc-env=JJ_VERSION={}+{}-{}",
+            version,
+            maybe_date
+                .map(|d| d.format("%Y%m%d").to_string())
+                .unwrap_or_else(|| "dateunknown".to_string()),
+            git_hash
+        );
     } else {
         println!("cargo:rustc-env=JJ_VERSION={}", version);
     }
 
     Ok(())
 }
 
-fn get_git_hash() -> Option<String> {
-    if let Some(nix_hash) = std::env::var("NIX_JJ_GIT_HASH")
+/// Convert a string with a unix timestamp to a date
+fn timestamp_to_date(ts_str: &str) -> Option<DateTime<Utc>> {
+    ts_str
+        .parse::<i64>()
         .ok()
-        .filter(|s| !s.is_empty())
-    {
-        return Some(nix_hash);
+        .and_then(|ts| DateTime::<Utc>::from_timestamp(ts, 0))
+}
+
+/// Return the UTC committer date (maybe) and the git hash
+fn get_git_timestamp_and_hash() -> Option<(String, Option<DateTime<Utc>>)> {
+    if let (Ok(nix_hash), maybe_nix_timestamp) = (
+        std::env::var("NIX_JJ_GIT_HASH"),
+        std::env::var("NIX_JJ_GIT_TIMESTAMP"),
```

Comment:
this is missing `rerun-if-env-changed` https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-env-changed

Reply:
you have it for `NIX_JJ_GIT_HASH` above, but not for `NIX_JJ_GIT_TIMESTAMP`. might make sense to move the println into this function, or add a wrapper which prints rerun every time you try to access the variable.